src/wallet/rpcwallet.cpp

```diff
@@ -2845,6 +2845,216 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
+    // Sign what we can:
+    bool fComplete = SignPartialTransaction(psbtx, pwallet, nHashType);
+    psbtx.sanitize_for_serialization();
+
+    UniValue result(UniValue::VOBJ);
+    CDataStream ssTx(SER_NETWORK, PROTOCOL_VERSION);
+    if (fComplete && (request.params[2].isNull() || (!request.params[2].isNull() && !request.params[2].get_bool()))) {
+        ssTx << psbtx.tx;
+    } else {
+        ssTx << psbtx;
+    }
+    result.push_back(Pair("hex", HexStr(ssTx.begin(), ssTx.end())));
+    result.push_back(Pair("complete", fComplete));
+
+    return result;
+}
+
+UniValue createpsbtfromraw(const JSONRPCRequest& request)
+{
+    CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+                            "createpsbtfromraw \"hexstring\"\n"
+                            "\nCreates a transaction in the Partially Signed Transaction format from a\n"
+                            "raw transaction that was funded with inputs from our wallet, typically from\n"
+                            "using fundrawtransaction.\n"
+                            "\nArguments:\n"
+                            "1. \"hexstring\"           (string, required) The hex string of the raw transaction\n"
+                            "\nResult:\n"
+                            "{\n"
+                            "  \"hex\":       \"value\", (string)  The resulting raw transaction (hex-encoded string)\n"
+                            "}\n"
+                            "\nExamples:\n"
+                            "\nCreate a transaction with no inputs\n"
+                            + HelpExampleCli("createrawtransaction", "\"[]\" \"{\\\"myaddress\\\":0.01}\"") +
+                            "\nAdd sufficient unsigned inputs to meet the output value\n"
+                            + HelpExampleCli("fundrawtransaction", "\"rawtransactionhex\"") +
+                            "\nMake PSBT to sign elsewhere\n"
+                            + HelpExampleCli("createpsbtfromraw", "\"fundedtransactionhex\"")
+                            );
+
+    RPCTypeCheck(request.params, {UniValue::VSTR});
+
+    CMutableTransaction mtx;
+
+    // Decode the transaction
+    if (!DecodeHexTx(mtx, request.params[0].get_str(), true))
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+
+    // Stuff to pass into PartiallySignedTransaction
+    std::map<uint160, CScript> redeem_scripts;
+    std::map<uint256, CScript> witness_scripts;
+    std::map<COutPoint, CTransaction> prev_txs;
+    std::map<COutPoint, CTxOut> witness_utxos;
+
+    // Get all of the previous transactions
+    for (unsigned int i = 0; i < mtx.vin.size(); ++i) {
+        CTxIn txin = mtx.vin[i];
+
+        uint256 hash = txin.prevout.hash;
+
+        if (!pwallet->mapWallet.count(hash)) {
```

Comment:
This is problematic for coinjoin, as no single participant can be expected to have all information about all inputs.

Instead, how about making this RPC accept either a serialized PSBT or raw transaction (they're never ambiguous, right?), add the information where it can, and then output a PSBT? Another participant can then take the PSBT, and 'extend' it using the same RPC.

Perhaps they can be completely separated too... one RPC that converts a raw transaction to a PSBT (which is purely utility), and another that 'augments' a PSBT with wallet-specific information.